src/components/pages/UserFormModal/UserFormModal.tsx

```diff
@@ -1,79 +1,119 @@
+        status: props.modalState.user?.status || Status.active,
       },
     };
   }
 
-  closeButton = () => {
-    this.props.handleNotification("", "");
-    this.props.onClose();
+  handleEditModalSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    editUser(this.state.user)
+      .then((res) => {
+        const errorData = res.response?.data[0];
+        const message = errorData
+          ? `${errorData.field} ${errorData.message}`
+          : "User updated successfully";
+        if (errorData) {
+          this.props.handleNotification(message, MessageType.error);
+          return;
+        }
+        this.props.handleNotification(message, MessageType.success);
+        this.props.handleUserAddOrEdit();
+        this.props.onCloseModal();
+      })
+      .catch((error) => {
+        this.props.handleNotification(
+          "Failed to update user",
+          MessageType.error
+        );
+      });
   };
 
-  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const { name, value } = e.target;
-    this.setState((prevState) => ({
-      user: {
-        ...prevState.user,
-        [name]: value,
-      },
-    }));
+  handleAddUserModalSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    addUser(this.state.user)
+      .then((res) => {
+        const errorData = res.response?.data[0];
+        const message = errorData
+          ? `${errorData.field} ${errorData.message}`
+          : "User added successfully";
+        if (errorData) {
+          this.props.handleNotification(message, MessageType.error);
+          return;
+        }
+        this.props.handleNotification(message, MessageType.success);
+        this.props.handleUserAddOrEdit();
+        this.props.onCloseModal();
+      })
+      .catch((error) => {
+        this.props.handleNotification("Failed to add user", MessageType.error);
+      });
   };
 
-  handleSubmit = async (e: React.FormEvent) => {
-    let status: number;
+  handleSubmit = (e: React.FormEvent) => {
     e.preventDefault();
+    this.props.modalState.type === ActionType.add
+      ? this.handleAddUserModalSubmit(e)
+      : this.handleEditModalSubmit(e);
+  };
 
-    try {
-      status = await addUser(this.state.user);
-
-      if (status === 201) {
-        this.props.handleNotification("User added successfully", "success");
-        this.props.handleUserAdd();
-        this.props.onClose();
-      } else if (status === 422) {
-        this.props.handleNotification("Email already exists", "error");
-      }
-    } catch (error) {
-      this.props.handleNotification("Failed to add user", "error");
-    }
+  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    this.setState((prevState) => ({
+      user: { ...prevState.user, [name]: value },
+    }));
   };
 
   render() {
     return (
       <>
-        <div className="add-user-form">
+        <div className="edit-user-form">
```

Comment:
```suggestion
        <div className="user-form">
```
use generic classnames for add and edit form type